src/wallet/rpc/backup.cpp

```diff
@@ -1784,34 +1784,59 @@ RPCHelpMan listdescriptors()
 
     LOCK(wallet->cs_wallet);
 
-    UniValue descriptors(UniValue::VARR);
     const auto active_spk_mans = wallet->GetActiveScriptPubKeyMans();
+
+    struct WalletDescInfo {
+        std::string descriptor;
+        uint64_t creation_time;
+        bool active;
+        std::optional<bool> internal;
```

Comment:
this field doesn't need to be an optional. It will always have a value.

Reply:
when a descriptor is not active we can't determine whether it's internal or not

Reply:
Indeed, this is why `IsInternalScriptPubKeyMan` returns `std::optional<bool>`. We list both active and inactive descriptors.

Reply:
k gotcha, nvm then.